hub/api/dataset.py

```diff
@@ -21,68 +19,77 @@
+                Can be a Hub cloud path of the form hub://username/datasetname. To write to Hub cloud datasets, ensure that you are logged in to Hub (use 'activeloop login' from command line)
+                Can be a s3 path of the form s3://bucketname/path/to/dataset. Credentials are required in either the environment or passed to the creds argument.
+                Can be a local file system path of the form ./path/to/dataset or ~/path/to/dataset or path/to/dataset.
+                Can be a memory path of the form mem://path/to/dataset which doesn't save the dataset but keeps it in memory instead. Should be used only for testing as it does not persist.
+            read_only (bool): Opens dataset in read only mode if this is passed as True. Defaults to False.
+                Datasets stored on Hub cloud that your account does not have write access to will automatically open in read mode.
             index (Index): The Index object restricting the view of this dataset's tensors.
             memory_cache_size (int): The size of the memory cache to be used in MB.
             local_cache_size (int): The size of the local filesystem cache to be used in MB.
-            storage (StorageProvider, optional): The storage provider used to access
-                the data stored by this dataset. If this is specified, the path given is ignored.
+            creds (dict, optional): A dictionary containing credentials used to access the dataset at the path.
+                This takes precedence over credentials present in the environment. Currently only works with s3 paths.
+                It supports 'aws_access_key_id', 'aws_secret_access_key', 'aws_session_token', 'endpoint_url' and 'region' as keys.
+            storage (StorageProvider, optional): The storage provider used to access the dataset.
+                Use this if you want to specify the storage provider object manually instead of using a path to generate it.
 
         Raises:
             ValueError: If an existing local path is given, it must be a directory.
-            UserWarning: Both path and storage should not be given.
-            UnsupportedModeError: If mode is not any of the options listed above.
+            ImproperDatasetInitialization: Exactly one argument out of 'path' and 'storage' needs to be specified.
+                This is raised if none of them are specified or more than one are specifed.
+            InvalidHubPathException: If a Hub cloud path (path starting with hub://) is specified and it isn't of the form hub://username/datasetname.
+            AuthorizationException: If a Hub cloud path (path starting with hub://) is specified and the user doesn't have access to the dataset.
+            PathNotEmptyException: If the path to the dataset doesn't contain a Hub dataset and is also not empty.
         """
-        if not mode in SUPPORTED_MODES:
-            raise UnsupportedModeError(mode)
+        if creds is None:
+            creds = {}
```

Comment:
Is there a reason why this is not assigned in the constructor?

Reply:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments